openmc/mixin.py

```diff
@@ -0,0 +1,24 @@
+import numpy as np
+
+
+class EqualityMixin(object):
+    """A Class which provides generic __eq__ and __ne__ functionality which
+    can easily be inherited by downstream classes.
+    """
+
+    def __eq__(self, other):
+        eqval = True
+        if isinstance(other, type(self)):
+            for key, value in self.__dict__.items():
+                if key in other.__dict__:
+                    if not np.array_equal(value, other.__dict__.get(key)):
+                        eqval = False
```

Comment:
Rather than store `eqval`, you can just prematurely `return False` which would avoid checks on any further attributes. Same goes for the other assignments to `eqval`. Then just have the final statement be `return True`.